### DISCUSSION TITLE
tutorials/animation/playing_videos

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#127] [User-contributed notes]
# tutorials/animation/playing_videos

Godot supports video playback with the VideoStreamPlayer node. Supported playback formats: The only supported format in core is Ogg Theora(not to be confused with Ogg Vorbis audio). It's possible f...

https://docs.godotengine.org/en/stable/tutorials/animation/playing_videos.html

**Comment by NicholasDewberryOfficial** [comment]
We ought to remove the handbrake reference from this page. It's incapable of converting video from mp4 to ogv, and this page is misleading people.

https://www.reddit.com/r/godot/comments/1ddo256/how_do_i_convert_an_mp4_video_to_ogv_using/

https://www.reddit.com/r/godot/comments/1408he6/convert_from_mp4_to_ogv_for_godot_without_using/ (comment at the bottom saying they couldnt use ffmpeg)

Instead, I'd recommend adding: https://www.shutterencoder.com/ to the documentation. It's a GUI for FFmpeg and is great for beginner developers who just want to downsize videos without any complex legwork.

**Reply by Calinou** [reply]
[This documentation change was merged in `master`](https://github.com/godotengine/godot-docs/pull/10732) but wasn't cherry-picked to `stable` yet.

**Comment by JGameMaker92** [comment]
I would like to see a way to make the VideoStreamPlayer play in the editor. Currently there's not even a preview of the video in the editor either, it's just an empty invisible box as a placeholder for where the video would be when you press play. It's hard to work with it when you can't even see it fully.

**Reply by Calinou** [reply]
Try attaching a script to the VideoStreamPlayer node, then save and use **Scene > Reload Saved Scene**:

```gdscript
@tool
extends VideoStreamPlayer


func _ready():
	play()
```

I haven't tested this, but it should logically play the video while in the editor (including audio).

**Reply by JGameMaker92** [reply]
Oh I guess that does work. But yeah, only after you reload the scene. It still wouldn’t hurt to have a built in video preview though. It feels kinda lacking without it

**Comment by swAAn01** [comment]
In case anyone here is having issues using FFmpeg to convert to .ogv in Windows, you're not alone. As it turns out, there's a bug that in some cases creates artifacting if you transcode in Windows. The workaround is really simple, you just have to do it in a Linux environment. Even a virtual machine or `wsl` will work just fine.

**Reply by Calinou** [reply]
Can you link to a FFmpeg bug report related to this? I don't remember hearing about it before.

**Reply by berarma** [reply]
Yes, there was a bug in Windows 64bits builds that is still present in old builds. Newer builds from the official website should be fixed. The documentation recommends using recent builds because they will have the latest fixes. Did you find a recent bugged build?

**Reply by berarma** [reply]
> Can you link to a FFmpeg bug report related to this? I don't remember hearing about it before.

I found out about this a couple months ago and it should be already fixed.

For reference:

https://github.com/BtbN/FFmpeg-Builds/pull/513

https://github.com/GyanD/codexffmpeg/issues/196

https://github.com/GyanD/codexffmpeg/issues/200
